### PR TITLE
refactor(files): upgrade component - INNO-588

### PR DESCRIPTION
**Pull request by emeryro**
upgrade of files component.
It contains file download and some media file display.

### CODE REVIEW

File: framework/components/ecl-files/_files.scss [hunk 1]
```diff
@@ -1,289 +1,193 @@
...
+/**
+ * File
+ * @define file
+ */
 
-/*
-File download default
-*/
+// Temporary hack, to be integrated in reset
+/* stylelint-disable-next-line */
```
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: Let's not forget to remove those lines.

File: framework/components/ecl-files/_files.scss [hunk 2]
```diff
@@ -1,289 +1,193 @@
...
 
-@include ecl-media-breakpoint-up(sm) {
-  .file__icon {
-    width: auto;
-    padding: 0;
-  }
+.ecl-file__translations-download {
+  align-self: flex-start;
+  font-weight: normal;
+  margin-left: -1rem;
```
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: How does it come that there is a negative margin?

File: framework/components/ecl-files/_files.scss [hunk 3]
```diff
@@ -1,289 +1,193 @@
...
-Style guide: file.5
-*/
+// Image and video file
 
-.file--video {
-  padding: 0;
-  background-color: transparent;
+.ecl-file--image,
+.ecl-file--video {
+  display: table;
```
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: Ugh...

File: framework/components/ecl-files/ecl-files.twig [hunk 4]
```diff
@@ -1,112 +1,147 @@
...
-  <div class="file__metadata">
-    <span class="file__title">File title example</span>
-    <div class="file__info">
-      <span class="file__content-language">English</span> (213.25 kB - PDF - 4 pages)
+{% if variant is defined %}
+
+  {% if variant == "translations" %}
+
+<!-- Translations -->
+<div class="ecl-file">
```
Comment (yhuard) [inline framework/components/ecl-files/ecl-files.twig]: This `div` could be a `section`

File: framework/components/ecl-files/_files.scss [hunk 5]
```diff
@@ -1,289 +1,193 @@
...
-  padding: 0.722em 0;
-  background-color: transparent;
-
-  .file__btn {
-    border: none;
-
-    &:hover {
-      color: $ecl-color-shade;
-    }
+    @include ecl-icon('download');
```
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: Please increase the margin (button icons have `margin-left: 0.8em;`)

File: framework/components/ecl-files/_files.scss [hunk 6]
```diff
@@ -1,289 +1,193 @@
...
   float: left;
-  height: 2.556em;
-  // We remove on the left (as the image has some whitespace) as it causes
-  // a wrong looking margin. Then we add back the margin on the right, to have
-  // more spacing between those fields.
-  margin-right: 0.1em;
-  margin-left: -0.315em;
-  padding: 0;
-  text-align: center;
+  margin-top: map-get($ecl-spacing, 'xxxs');
```
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: Why did you add a margin here? Wasn't there before

File: framework/components/ecl-files/_files.scss (hunk 3 above)
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: BTW it is not very responsive

File: framework/components/ecl-files/_files.scss (hunk 2 above)
Comment (emeryro) [inline framework/components/ecl-files/_files.scss]: That's because it is a button without borders (modifier "secondary"), which comes with a padding. So to align the text I had to add some negative margin

File: framework/components/ecl-files/ecl-files.twig (hunk 4 above)
Comment (emeryro) [inline framework/components/ecl-files/ecl-files.twig]: well I kept a similar markup between all modifiers.
I could set is as a section, but should we also use section for the other ones?

File: framework/components/ecl-files/_files.scss (hunk 2 above)
Comment (yhuard) [inline framework/components/ecl-files/_files.scss]: Makes sense

File: framework/components/ecl-files/ecl-files.twig (hunk 4 above)
Comment (yhuard) [inline framework/components/ecl-files/ecl-files.twig]: Yes, I just pointed this one as an example :)

File: framework/components/ecl-files/_files.scss (hunk 6 above)
Comment (emeryro) [inline framework/components/ecl-files/_files.scss]: I find display better this way :)

File: framework/components/ecl-files/_files.scss (hunk 5 above)
Comment (emeryro) [inline framework/components/ecl-files/_files.scss]: which margin are you referring to?

**Comment by emeryro** [comment]
Modifications have been done, ready for review

File: framework/components/ecl-files/package.json [hunk 7]
```diff
@@ -9,7 +9,6 @@
   "license": "EUPL-1.1",
   "peerDependencies": {
     "@ec-europa/ecl-buttons": "^0.1.0",
-    "@ec-europa/ecl-fields": "^0.1.0",
```
Comment (kalinchernev) [inline framework/components/ecl-files/package.json]: The `private` prop can be removed, and the usual public props to be added.

File: framework/components/ecl-files/ecl-files.config.js [hunk 8]
```diff
@@ -1,6 +1,91 @@
...
+        },
+      },
+    },
+    {
+      name: 'links',
+      label: 'Links to files',
+      context: {
+        variant: 'links',
+        links: [
+          { title: 'First file', icon: 'file', href: '#', type: 'pdf' },
```
Comment (kalinchernev) [inline framework/components/ecl-files/ecl-files.config.js]: I personally find the previous file names a bit more user-friendly, i.e. `Generic external file link`, `Link to a .pdf file`, etc.

File: framework/components/ecl-files/ecl-files.twig [hunk 9]
```diff
@@ -1,112 +1,144 @@
...
     </div>
   </div>
-  <a href="#" class="btn file__btn btn--condensed is-internal">
-    Download<span class="ecl-sr-only">(213.25 kB - PDF - 4 pages)</span>
-  </a>
-</div>
-
-<!-- Image -->
-<div style="width: 450px;"> <!-- only required for the style guide -->
-  <div class="file file--image">
```
Comment (kalinchernev) [inline framework/components/ecl-files/ecl-files.twig]: There's also an implementation of the file in `ecl-banners.twig` which has:
```
      <div class="file file--video">
        <div class="file__container">
          <iframe src="http://ec.europa.eu/avservices/play.cfm?ref=I101631" allowfullscreen="allowfullscreen" scrolling="no"></iframe>
        </div>
        <span class="banner__caption">
          Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
        </span>
      </div>
```

Comment (emeryro) [inline framework/components/ecl-files/ecl-files.twig]: yes, indeed. We will have to remember this when upgrading the banner component.
I added a comment in dedicated ticket

**Comment by emeryro** [comment]
Modifications have been done.
@kalinchernev @yhuard could you do a final review and give approval?

File: test/functional/files.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      context('with plain state', () => {
+        it('should match the reference screenshot', () => {
+          const screenshots = browser.checkDocument({
+            name: `files/${variant}`,
+          });
+          expect(screenshots).to.matchReference();
+        });
+
+        it('should be accessible', () => {
+          const a11yReport = browser.runAxeCore('ecl-files').value;
```
Comment (yhuard) [inline test/functional/files.js]: Beware! It should be `'ecl-file'` and not `'ecl-files'`. That's the reason why the a11y tests are not run

**Comment by yhuard** [comment]
Some screenshots are missing and other tests are failing

**Comment by yhuard** [comment]
I've been investigating and found out that the video variants cause troubles. Maybe because of Flash, maybe because of the async content. Anyway, I've deactivated those tests and updated the screenshots. Now travis can run again

File: framework/components/ecl-files/ecl-files.twig [hunk 11]
```diff
@@ -1,112 +1,144 @@
...
+<!-- Images -->
+<section class="ecl-file ecl-file--image">
+  <img class="ecl-file__image" src="{{ src }}" alt="{{ alt }}">
+  <div class="ecl-file__caption">{{ caption }}</div>
+</section>
+
+  {% elseif variant == "Videos" %}
+
+<!-- Videos -->
+<section class="ecl-file ecl-file--video">
```
Comment (yhuard) [inline framework/components/ecl-files/ecl-files.twig]: It would be much appreciated to keep the aspect ratio. See https://css-tricks.com/snippets/sass/maintain-aspect-ratio-mixin/ :)

Comment (emeryro) [inline framework/components/ecl-files/ecl-files.twig]: Ratio utilities have been added, and applied to video files

**Comment by yhuard** [comment]
Much better!
